kiwi/lib/search.py

```diff
@@ -0,0 +1,655 @@
+            )
+            base_config_dict['system']['model']['outputs'][
+                'sentence_loss_weight'
+            ] = sentence_loss_weight
+            search_values['sentence_loss_weight'] = sentence_loss_weight
+
+        # The same logic applies here: when search whether to include the sentence level
+        #   objective (`search_hter`) then we will only want to search the
+        #   `sentence_loss_weight` when we are actually including the sentence level
+        #   objevtive (`hter = True`). See the above definition of the variable `hter`.
+        if (
+            word_level_config.get('target')
+            and self.config.options.search_hter
+            and self.config.options.sentence_loss_weight
+        ):
+            if hter:
+                sentence_loss_weight = get_suggestion(
+                    trial,
+                    'sentence_loss_weight',
+                    self.config.options.sentence_loss_weight,
+                )
+                base_config_dict['system']['model']['outputs'][
+                    'sentence_loss_weight'
+                ] = sentence_loss_weight
+                search_values['sentence_loss_weight'] = sentence_loss_weight
+
+        if self.config.options.class_weights and word_level_config:
+            # if self.config.options.search_word_level:
+            #     if not word_level:
+            #         # Word level disabled; nothing to search over
+            #         pass
+            if (
+                self.config.options.class_weights.target_tags
+                and word_level_config['target']
+            ):
+                class_weight_target_tags = get_suggestion(
+                    trial,
+                    'class_weight_target_tags',
+                    self.config.options.class_weights.target_tags,
+                )
+                base_config_dict['system']['model']['outputs']['word_level'][
+                    'class_weights'
+                ]['target_tags'] = {const.BAD: class_weight_target_tags}
+                search_values['class_weight_target_tags'] = class_weight_target_tags
+            if self.config.options.class_weights.gap_tags and word_level_config['gaps']:
+                class_weight_gap_tags = get_suggestion(
+                    trial,
+                    'class_weight_gap_tags',
+                    self.config.options.class_weights.gap_tags,
+                )
+                base_config_dict['system']['model']['outputs']['word_level'][
+                    'class_weights'
+                ]['gap_tags'] = {const.BAD: class_weight_gap_tags}
+                search_values['class_weight_gap_tags'] = class_weight_gap_tags
+            if (
+                self.config.options.class_weights.source_tags
+                and word_level_config['source']
+            ):
+                class_weight_source_tags = get_suggestion(
+                    trial,
+                    'class_weight_source_tags',
```

Comment:
you can collapse these 3 ifs with a loop that does:

```
For side in ['target', 'gaps', 'source']:
  if config.options.class_weights.side and world_level_config['side']:
     etc
```

Reply:
Agree